domain/model/bootstrap/bootstrap.go

```diff
@@ -30,6 +39,17 @@ func CreateModel(
 		if err := args.Validate(); err != nil {
 			return fmt.Errorf("model creation args: %w", err)
 		}
+
+		agentVersion := args.AgentVersion
+		if args.AgentVersion == version.Zero {
+			agentVersion = jujuversion.Current
+		}
+
+		if agentVersion.Compare(jujuversion.Current) != 0 {
+			return fmt.Errorf("%w %q during bootstrap", modelerrors.AgentVersionNotSupported, agentVersion)
+		}
+		args.AgentVersion = agentVersion
```

Comment:
I'm not sure why we even need AgentVersion here. I guess we use CreateModelArgs for hosted models later. We could error if AgentVersion is even set at all. I guess the check here makes it more robust.

Reply:
We probably don't need it long term. But at the moment the way our bootstrap process works is that the client is making a lot of decisions around what agent version number we insert into state when starting up. Harry's work around snapping the controller is going to deal mostly with this so I have decided to leave it alone for the time being.

Ideally the controller bootstrap process can take ownership of it's version again and not have the client drive it.

Reply:
As discussed f2f, we don't need this at bootstrap. The client uses the requested agent version to pull the relevant tarball from which jujud is unpacked - thereafter, we can just use the jujuversion.Current and not pass in AgentVersion anywhere at all for the rest of the bootstrap logic.
Let's add a TODO to clean this up.